Packages/com.unity.inputsystem/InputSystem/InputSystem.cs

```diff
@@ -3072,6 +3072,27 @@ internal static void OnPlayModeChange(PlayModeStateChange change)
                         settings.OnChange();
                     }
 
+                    // reload input action assets marked as dirty from disk
+                    if (s_TrackedDirtyAssets == null)
+                        return;
+
+                    foreach (var assetGuid in s_TrackedDirtyAssets)
+                    {
+                        var assetPath = AssetDatabase.GUIDToAssetPath(assetGuid);
+
+                        if (string.IsNullOrEmpty(assetPath))
+                        {
+                            Debug.LogWarning($"Input action asset change tracking was tracking asset with GUID '{assetGuid}' but it wasn't found in the asset " +
+                                             $"database on exiting play mode. If you have added bindings to this asset during this play mode session, you may need " +
+                                             $"to reimport the asset from disk to remove the added bindings from memory.");
```

Comment:
Probably better to just silently ignore. Asset may have been deleted in the meantime so it no longer being found would be expected and perfectly fine. If the asset popped up somewhere else (say, someone did a `git checkout` that happened to change the GUID in the .meta file), we're also good as the importer has brought in a clean state.